### ISSUE TITLE
ECS ServiceName attribute

### ISSUE THREAD

**Issue by pmm-sumo** [open-telemetry/opentelemetry-specification#1836]
**What are you trying to achieve?**

[AWS ECS Semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/349c9ab33ba012b375c2a4f90d1e6391dc2d488e/specification/resource/semantic_conventions/cloud_provider/aws/ecs.md) define a number of attributes related to ECS. 

It misses an ability to describe [Service Name](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecs-service.html). Perhaps an attribute such as `aws.ecs.service.name` could be introduced?

**Comment by Oberon00** [comment]
Can you use the general `service.name` for that? Or would you set something different there?

**Comment by pmm-sumo** [comment]
> Can you use the general `service.name` for that? Or would you set something different there?

I believe those two values are set independently (one when creating AWS service, the other when configuring instrumentation) so they could hold different values

**Comment by Oberon00** [comment]
Nitpick: service.name is set when configuring the application that is using OpenTelemetry, not the instrumentation.

I wonder if it would be best practice to set this to the same value as the ECS service name, as they seem to conceptually be the same thing, even if they might technically differ. I.e., should it be considered misconfiguration if the service.name is configured differently from the ECS service name, or are there valid reasons for making them different.

Or am I barking up the wrong tree here and the ECS service name is a completely different thing that just happens to have the same name?

CC @willarmiros

**Comment by willarmiros** [comment]
> Or am I barking up the wrong tree here and the ECS service name is a completely different thing that just happens to have the same name?

Yes unfortunately this appears to be a naming collision, ECS Services are an offering from ECS, see: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs_services.html

@pmm-sumo I considered adding such an attribute initially, but decided against it because initially we were assuming the collector would only be run as a sidecar rather than a daemon as part of an ECS Service. Even though I'm still not sure we officially support operating the collector as a service rather than a sidecar, that shouldn't be blocking adding service-related attributes (like `aws.ecs.service.name`) to the semantic conventions, so I'd be in favor of this.

**Comment by RaphaelManke** [comment]
Hey guys, I am wondering if there is anything blocking adding the service name to the semantic conventions, or are you open to accept the contribution to add this attribute? Background of my request is that I noticed that this attribute is used for the [ECS metrics receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/19744), but not for the resource detector.

And if I read the conversations about the resource detector correctly, it is mainly because the service name is not added in the semantic conventions yet. But from this thread, I read that it's actually okay to add it to the semantic convention. So I'm wondering if it's just something that needs to be done, or is there still an open discussion if it should be added? 

https://github.com/open-telemetry/semantic-conventions/issues/1199
